blueocean-dashboard/src/main/js/components/StageRestartLink.tsx

```diff
@@ -74,12 +73,16 @@ class StageRestartLink extends React.Component<Props, State> {
     }
 
     render() {
-        return (
-            <a className="restart-stage" onClick={() => this._onRestartStageClick()}>
-                <RestartStageIcon />
-                <span>{this.props.t('rundetail.logToolbar.restartStage', { 0: this.props.title })}</span>
-            </a>
-        );
+        if (Security.permit(this.props.pipeline).start()) {
+            return (
+                <a className="restart-stage" onClick={() => this._onRestartStageClick()}>
+                    <RestartStageIcon />
+                    <span>{this.props.t('rundetail.logToolbar.restartStage', { 0: this.props.title })}</span>
+                </a>
+            );
+        } else {
+            return false;
```

Comment:
won't this render "false" or something? I don't think you need the fallthrough return statement

Reply:
Nah react won't render `false` or `undefined`. You have to be careful about your truthyness tho, if you've got a comparison that can return 0 rather than false it *will* render that.

Reply:
yup, in this case there is no danger of things being rendered incorrectly, it either shows the link or it doesn't render any tag or content